torch/csrc/Stream.cpp

```diff
@@ -37,9 +77,28 @@ static PyObject* THPStream_pynew(
   }
 
   THPStream* self = (THPStream*)ptr.get();
-  self->stream_id = stream_id;
-  self->device_index = device_index;
-  self->device_type = device_type;
+
+  // If torch.Stream is not created from existing Stream, then create a new one.
+  // It requires other device backends override getNewStream method. How the new
+  // stream is created is backend specific. Backend should be able to correctly
+  // manage the lifetime of streams.
+  c10::Stream stream(c10::Stream::DEFAULT, c10::Device(c10::DeviceType::CPU));
```

Comment:
Can we avoid creating this object by any chance? Making this an optional<Stream> would work at least.

Reply:
Good catch ! we should do that.